java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java

```diff
@@ -221,18 +222,27 @@ public void visitMethodInvocation(MethodInvocationTree tree) {
       typeParamTypes = getParameterTypes(tree.typeArguments());
     }
     Resolve.Resolution resolution = resolveMethodSymbol(methodSelect, methodEnv, argTypes, typeParamTypes);
-    JavaSymbol symbol = resolution.symbol();
+    JavaSymbol symbol;
+    JavaType returnType;
+    if(resolution == null) {
+      returnType = symbols.deferedType(mit);
+      symbol = Symbols.unknownSymbol;
+    } else {
+      symbol = resolution.symbol();
+      returnType = resolution.type();
+    }
     mit.setSymbol(symbol);
-    JavaType returnType = resolution.type();
-    if(resolution.symbol().isMethodSymbol()) {
+    if(symbol.isMethodSymbol()) {
       MethodJavaType methodType = (MethodJavaType) resolution.type();
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'resolution' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)